page_analyzer/app.py

```diff
@@ -1,8 +1,58 @@
-from flask import Flask, render_template
+import os
+from typing import Any
+
+from dotenv import load_dotenv
+from flask import Flask, flash, redirect, render_template, request, url_for
+from page_analyzer.utils.db import (
+    check_url_existence,
+    create_url,
+    get_url_context,
+    get_urls_data,
+)
+from page_analyzer.utils.url import normalize, validate
+
+load_dotenv()
+
 
 app = Flask(__name__)
+app.secret_key = os.getenv("SECRET_KEY")
 
 
 @app.route("/")
-def hello_world() -> str:
+def index() -> str:
     return render_template("index.html")
+
+
+@app.route("/url", methods=["POST"])
+def add_url() -> Any:
+    url_name = request.form["url"]
+    errors = validate(url_name)
+    if errors:
+        flash(errors["name"], "danger")
+        return render_template("index.html")
+    url_name = normalize(url_name)
+    url_check_result = check_url_existence(url_name)
+    if url_check_result:
+        url_id = url_check_result
```

Comment:
wdyt about this option?

```suggestion
    url_id = get_url_by_name(url_name)
    if url_id:
```